pulser/register.py

```diff
@@ -53,6 +55,7 @@ def __init__(self, qubits: Mapping[Any, ArrayLike]):
             raise ValueError("All coordinates must be specified as vectors of"
                              " size 2 or 3.")
         self._coords = coords
+        self._mag_field = mag_field
```

Comment:
Let's make the magnetic field a "property".  So it will have a `@property` decorator and a setter function (that corresponds in line 459)